FileManager/Core/Services/DirectoryService.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System.Diagnostics;
+using FileManager.Core.Services.Interfaces;
+using Serilog;
+
+namespace FileManager.Core.Services
+{
+    public class DirectoryService : IDirectoryService, ISingleton<DirectoryService>
+    {
+        private readonly string _error = "Internal error!";
+
+        /// <summary> Копирует содержимое директории </summary>
+        /// <param name="sourcePath">путь источника</param>
+        /// <param name="destPath">путь цели</param>
+        public void Copy(string sourcePath, string destPath)
+        {
+            var dirSource = new DirectoryInfo(sourcePath);
+            var dirDest = new DirectoryInfo(destPath);
+
+            CopyAll(dirSource, dirDest);
+        }
+
+        private void CopyAll(DirectoryInfo source, DirectoryInfo target)
+        {
+            try
+            {
+                Directory.CreateDirectory(target.FullName);
+
+                foreach (var fileInfo in source.GetFiles())
```

Comment:
Если задача стоит просто перебрать файлы/директории, то не надо формировать массив...
```suggestion
                foreach (var fileInfo in source.EnumerateFiles())
```